lib/src/widgets/picker.dart

```diff
@@ -200,34 +300,48 @@ class _PickerState<Item> extends State<Picker<Item>> {
               shrinkWrap: true,
               separatorBuilder: (context, index) => widget.isSeparated
                   ? Divider(
-                      color: Theme.of(context).accentTextTheme.headline6!.backgroundColor!,
+                      color: Theme.of(context)
+                          .accentTextTheme
+                          .headline6!
+                          .backgroundColor!,
                       height: 1,
                     )
                   : const SizedBox(),
-              itemCount: items.isEmpty ? 0 : items.length,
+              itemCount: filteredItems.isEmpty ? 0 : filteredItems.length,
               itemBuilder: (context, index) => buildItem(index),
             ),
     );
   }
 
-  Widget buildItem(int index) {
-    /// don't show selected item in the list view
-    if (widget.items[widget.selectedAtIndex] == items[index] && !widget.isGridView) {
-      return const SizedBox();
-    }
+  Widget buildItem(int index, [bool? buildSelected]) {
+    final isSelected = buildSelected == true;
```

Comment:
I would say we keep them separated instead of having boolean comparisons everywhere, so it's more readable, besides, it's more scalable when separated, as whenever you want to change the selected item UI only, you won't have to go through all these boolean comparisons